tests/test_multiprocess.py

```diff
@@ -240,13 +246,26 @@ def test_child_process(self):
             self.template(["viztracer", "-o", os.path.join(tmpdir, "result.json"), "--subprocess_child", "child.py"],
                           expected_output_file=None)
             self.assertEqual(len(os.listdir(tmpdir)), 1)
+            with open(os.path.join(tmpdir, os.listdir(tmpdir)[0])) as f:
+                trace_events = json.load(f)["traceEvents"]
+                for entry in trace_events:
+                    if entry["name"] == "process_name":
+                        self.assertNotEqual(entry["args"]["name"], "MainProcess")
+                        break
+                else:
+                    self.fail("no process_name event found")
 
     def test_module(self):
         self.template(["viztracer", "-o", "result.json", "cmdline_test.py"],
                       expected_output_file="result.json",
                       expected_stdout=".*100 loops.*",
                       script=file_subprocess_module)
 
+    def test_code(self):
+        self.template(["viztracer", "-o", "result.json", "cmdline_test.py"],
+                      expected_output_file="result.json",
+                      script=file_subprocess_code)
```

Comment:
Let's also check the process name of the subprocess on this, as it's known - should be `python` I believe?

Reply:
Yes, the current process_name is `site-packages/viztracer/__main__.py` because `sys.argv` is processed in `VizUI.run()`, which is before the config.

Reply:
Having a process with a name `python` is almost as bad as `MainProcess`. Let's set the `process_name` in `run_code`. Update `self.init_kwargs` before create the `VizTracer` instance. The subprocess would be distinguishable with the distinct names.

Reply:
If setting `process_name` in `run_code`. For module, the `process_name` is module name. For run python file, the `process_name` is python file name. For run code by `-c`, the process_name is `-c`.

Reply:
Yeah `"-c"` is a bit weird. Let's do another check for `cmd_string` and if it's set, set the process name to `"python -c"`.